asn1tools/compiler.py

```diff
@@ -372,7 +376,7 @@ def compile_files(filenames,
 
     """
 
-    if cache_dir is None:
+    if cache_dir is None or not has_diskcache:
```

Comment:
Probably better to raise an exception with a nice error message (or similar) if a cache directory is given but diskcache is not installed.

Reply:
Agreed.